service/matching/task_queue_manager.go

```diff
@@ -809,20 +818,52 @@ func (c *taskQueueManagerImpl) RedirectToVersionedQueueForAdd(ctx context.Contex
 
 	if c.kind == enumspb.TASK_QUEUE_KIND_STICKY {
 		// In the sticky case we don't redirect, but we may kick off this worker if there's a newer one.
-		err := checkVersionForStickyAdd(data, buildId)
-		return c.taskQueueID, userDataChanged, err
+		unknownBuild, err := checkVersionForStickyAdd(data, buildId)
+		if err != nil {
+			return nil, nil, err
```

Comment:
Should we return `userDataChanged` here?

Reply:
I don't think we need to. None of the callers look at it if err is non-nil. As much as possible, I'd like to keep to the pattern of not returning both error and values. (I know I break that rule in the dlq PR, but it's only for errUserDataDisabled and we can remove it all when we get rid of the kill switch)